tests/lib/rules/no-unused-vars.js

```diff
@@ -1792,6 +2034,460 @@ _ => { _ = _ + 1 };
+            errors: [assignedError("b", [{ output: "", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "const { a: [[ b ]]} = foo;",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("b", [{ output: "", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "const [{ a: [{ b }]}] = foo;",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("b", [{ output: "", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "const { a: [{ b }], c} = foo; alert(c);",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("b", [{ output: "const {  c} = foo; alert(c);", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "function foo({ a: [{ b }]}) {} foo();",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [definedError("b", [{ output: "function foo() {} foo();", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "function foo({ a: [[ b ]]}) {} foo();",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [definedError("b", [{ output: "function foo() {} foo();", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "let a = foo, b = 'bar'; alert(b);",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("a", [{ output: "let  b = 'bar'; alert(b);", messageId: "removeVar", data: { varName: "a" } }])]
+        },
+        {
+            code: "let a = foo, b = 'bar'; alert(a);",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("b", [{ output: "let a = foo; alert(a);", messageId: "removeVar", data: { varName: "b" } }])]
+        },
+        {
+            code: "let { a } = foo, bar = 'hello'; alert(bar);",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("a", [{ output: "let  bar = 'hello'; alert(bar);", messageId: "removeVar", data: { varName: "a" } }])]
+        },
+        {
+            code: "let bar = 'hello', { a } = foo; alert(bar);",
+            languageOptions: { ecmaVersion: 2023 },
+            errors: [assignedError("a", [{ output: "let bar = 'hello'; alert(bar);", messageId: "removeVar", data: { varName: "a" } }])]
+        },
+        {
+            code: "import a from 'module';",
+            languageOptions: { ecmaVersion: 6, sourceType: "module" },
+            errors: [definedError("a", [{ output: "", messageId: "removeVar", data: { varName: "a" } }])]
+        },
+        {
+            code: "import * as foo from 'module'",
+            languageOptions: { ecmaVersion: 6, sourceType: "module" },
+            errors: [definedError("foo", [{ output: "", messageId: "removeVar", data: { varName: "foo" } }])]
+        },
```

Comment:
In those cases it seems more correct to keep the import and just drop the variable. So the suggestion output should be probably `import 'module';`.

Reply:
Done!